server/handler/check_run.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2018 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/google/go-github/github"
+	"github.com/palantir/go-githubapp/githubapp"
+	"github.com/pkg/errors"
+
+	"github.com/palantir/bulldozer/pull"
+)
+
+type CheckRun struct {
+	Base
+}
+
+func (h *CheckRun) Handles() []string {
+	return []string{"check_run"}
+}
+
+func (h *CheckRun) Handle(ctx context.Context, eventType, deliveryID string, payload []byte) error {
+	var event github.CheckRunEvent
+
+	if err := json.Unmarshal(payload, &event); err != nil {
+		return errors.Wrap(err, "failed to parse status event payload")
+	}
+
+	repo := event.GetRepo()
+	installationID := githubapp.GetInstallationIDFromEvent(&event)
+
+	ctx, logger := githubapp.PrepareRepoContext(ctx, installationID, repo)
+	logger.Debug().Msgf("Doing nothing since check_run action was %q instead of 'completed'", event.GetAction())
+
+	if event.GetAction() != "completed" {
+		logger.Debug().Msgf("Doing nothing since check_run action was %q instead of 'completed'", event.GetAction())
+		return nil
+	}
+	suite := event.GetCheckRun()
```

Comment:
Should this be `run` instead of `suite`?

Reply:
It was my original understanding that CheckRuns were needed because CheckSuites would only be sent once per commit, but now I'm not sure about that. I think both event types would work.

Reply:
Sorry, I wasn't clear - I only meant the variable name didn't match the method call. From my reading, I think `check_run` is the correct event.